pydoctor/epydoc/sre_parse36.py

```diff
@@ -504,15 +504,7 @@ def _parse_sub(source, state, verbose, nested):
             continue # check next one
         break
 
-    # check if the branch can be replaced by a character set
-    for item in items:
-        if len(item) != 1 or item[0][0] is not LITERAL:
-            break
-    else:
-        # we can store this as a character set instead of a
-        # branch (the compiler may optimize this even more)
-        subpatternappend((IN, [item[0] for item in items]))
-        return subpattern
+    # pydoctor: remove all optimizations for round-tripping issues
```

Comment:
@tristanlatr what do you need here exactly? I'm not super familiar with this code, and I am not sure what the relevant optimizations are. What would give you confidence the logic isn't broken, beyond the existing test suite?

Reply:
> Its not broken from a regex matching perspective, but it is from a regex colorizing perspective (since the roundtrip is impossible for some cases)

gotcha, this makes a bit more sense. My plate is honestly kind of full right now, so I don't know how much time I can dedicate to this. Maybe ping the mailing list to see if anyone is interested? Sustainable open source infrastructure is hard :(.